main.py

```diff
@@ -56,16 +57,31 @@ def pattern_match(patterns, source_list):
             task_names.add(matching)
     return sorted(list(task_names))
 
+def setup_example_logger(output_path, separator):
+    """Sets up a logger that will save each example and prediction."""
+    example_logger = logging.getLogger("examples")
+    filename = f"./outputs/examples{separator}{output_path}.jsonl"
+    formatter = logging.Formatter("%(message)s")
+    handler = logging.FileHandler(filename)
+    handler.setFormatter(formatter)
+    example_logger.addHandler(handler)
+    example_logger.setLevel(logging.INFO)
+
 
 def main():
-    args = parse_args()
+    os.makedirs("./outputs", exist_ok=True)
+    args = parse_args()    
```

Comment:
Should also utilize `output_path` for the `outputs/` dir.

Reply:
I am a bit doubtful if we can use `output_path` as the path to a folder.
From the [following](https://github.com/EleutherAI/lm-evaluation-harness/blob/b21c8f3dae152c8dbc93b001e1e2d6a6b93e6cf1/main.py#L76) lines, I think the `output_path` is directed to a file location which in the following lines it is going to open it. If we consider the `output_path` as a folder, we should also change these lines.
```
        if args.output_path:
            with open(args.output_path, "w") as f:
                f.write(dumped)
```

Reply:
So the full code has 

```
        os.makedirs(os.path.dirname(args.output_path), exist_ok=True)
```
inserted before `if args.output_path:`, I think we can use the `os.path.dirname` to acquire the directory path and use that for the sample logs.

Reply:
Great. Then, just in the case when the `output_path` is defined, the `example_logger` will be activated and logs the examples?